core/modules/filters/each.js

```diff
@@ -17,19 +18,23 @@ Export our filter function
 */
 exports.each = function(source,operator,options) {
 	var results = [],
-		values = {};
+		values = {},
+		list = "list" === operator.suffix;
 	source(function(tiddler,title) {
 		if(tiddler) {
-			var value;
-			if((operator.operand === "") || (operator.operand === "title")) {
-				value = title;
-			} else {
-				value = tiddler.getFieldString(operator.operand);
-			}
-			if(!$tw.utils.hop(values,value)) {
-				values[value] = true;
-				results.push(title);
-			}
+			var value,
+				field = operator.operand || "title";
+			$tw.utils.each(
+				list ?
```

Comment:
I think you should use an `if ... else` statement here. Understanding the code, imo would be easier, since there are several more conditionals.